api/src/app/modules/analytics-dashboard/analytics-dashboard.controller.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  Controller,
+  Get,
+  Query,
+  DefaultValuePipe,
+  ParseIntPipe,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags, ApiQuery } from '@nestjs/swagger';
+import { AnalyticsDashboardService } from '@api-modules/analytics-dashboard/analytics-dashboard.service';
+import { DateRangeRequest } from '@api-modules/analytics-dashboard/analytics-dashboard.dto';
+import {
+  ProjectCountByDistrictResponse,
+  ProjectCountByForestClientResponse,
+} from '@api-modules/project/project.dto';
+import {
+  PublicCommentCountByDistrictResponse,
+  PublicCommentCountByCategoryResponse,
+  PublicCommentCountByProjectResponse,
+} from '@api-modules/public-comment/public-comment.dto';
+
+@ApiTags('analytics-dashboard')
+@Controller('analytics-dashboard')
+export class AnalyticsDashboardController {
+  constructor(private readonly analyticsDashboardService: AnalyticsDashboardService) {}
+
+  /**
+   * Returns the total number of FOM projects submitted
+   * within the specified commenting open date range,
+   * excluding those with an INITIAL workflow status.
+   *
+   * @param query - DateRangeRequest containing startDate and endDate in 'YYYY-MM-DD' format
+   * @returns Number of FOM projects in the date range
+   */
+  @Get('project/count')
+  @ApiOperation({
+    summary: 'Get total number of FOMs submitted in a date range',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'Total number of FOMs',
+    type: Number,
+  })
+  async getProjectCount(@Query() query: DateRangeRequest): Promise<number> {
+    return this.analyticsDashboardService.getProjectCountByDate(
+      query.startDate,
+      query.endDate
+    );
+  }
+
+  /**
+   * Returns the number of FOM projects grouped by district
+   * within the specified commenting open date range,
+   * excluding those with an INITIAL workflow status.
+   *
+   * @param query - DateRangeRequest containing startDate and endDate in 'YYYY-MM-DD' format
+   * @returns An array of objects containing districtId, districtName, and projectCount
+   */
+  @Get('project/count-by-district')
+  @ApiOperation({
+    summary: 'Get total number of FOMs grouped by district in a date range',
+  })
+  @ApiResponse({
+    status: 200,
+    description: 'List of FOM project counts by district',
+    type: ProjectCountByDistrictResponse,
+    isArray: true,
+  })
+  async getProjectCountByDistrict(
+    @Query() query: DateRangeRequest
+  ): Promise<ProjectCountByDistrictResponse[]> {
+    return this.analyticsDashboardService.getProjectCountByDistrict(
+      query.startDate,
+      query.endDate
+    );
+  }
+
+  /**
+   * Returns the total number of unique forest clients who submitted FOMs
+   * within the specified commenting open date range,
```

Comment:
Not within the commenting open date range, but within the requested date range. If the logic is meant to be forest clients who published, then we can potentially check the commenting open date to see if it overlaps with the date range, although there might be a publish date that would be even more accurate to use.

Reply:
This one is for `number of FOMs submitted by each Forest Client`. Should it be **Returns the number of FOM projects by each forest client with a commenting open date within the sepcified date range, excluding those with an INITIAL workflow status**?